jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -896,15 +946,19 @@ def mos_niriss_parser(app, data_dir, instrument=None,
     with app.data_collection.delay_link_manager_update():
         for label, data in add_to_glue.items():
             app.add_data(data, label, notify_done=False)
+        _add_to_table(app, spec_labels_1d, "1D Spectra")
+        _add_to_table(app, spec_labels_2d, "2D Spectra")
+        _add_to_table(app, image_add, "Images")
 
         # We then populate the table inside this context manager as
         # _add_to_table does operations that also trigger link manager updates.
-        _add_to_table(app, source_ids, "Identifier")
-        _add_to_table(app, ras, "R.A.")
-        _add_to_table(app, decs, "Dec.")
-        _add_to_table(app, image_add, "Images")
-        _add_to_table(app, spec_labels_1d, "1D Spectra")
-        _add_to_table(app, spec_labels_2d, "2D Spectra")
-        _add_to_table(app, filters, "Filter/Grating")
+        meta_names = _get_source_identifiers(app, "2D Spectra")
+        _add_to_table(app, meta_names, "Identifier")
+        meta_filters = query_metadata_by_component(app, 'FILTER', "2D Spectra")
+        _add_to_table(app, meta_filters, "Filter/Grating")
```

Comment:
I think I'm not understanding the context that these would be called in vs the `add_to_table` calls up in `mos_meta_parser` - do they not conflict?

Reply:
Aha isn't that interesting... 😉 From the description:

> As a stretch goal, I also switch both the NIRISS and Generic/NIRspec parsers to using this new metadata parsing strategy. Notice how the metadata block at the bottom looks almost identical to the mos_metadata_parser used for the generic/nirspec strategy.... how interesting.... 😉 (Future PR)

The NIRISS parser here never used the `mos_meta_parser` and still doesn't here. It used to have its own parsing logic. In this PR, I gutted out the NIRISS meta logic and replaced it with code that looks STRANGELY close to the `mos_meta_parser` as you discovered here. _It's almost like we might be able to combine them soon~!_

Reply:
Got it 😄